pkg/registry/controller/rest.go

```diff
@@ -47,6 +47,12 @@ func (rcStrategy) NamespaceScoped() bool {
 func (rcStrategy) PrepareForCreate(obj runtime.Object) {
 	controller := obj.(*api.ReplicationController)
 	controller.Status = api.ReplicationControllerStatus{}
+	if len(controller.Spec.Selector) == 0 {
+		controller.Spec.Selector = controller.Spec.Template.Labels
```

Comment:
The template is optional. It could be nil. We're working on allowing it to be specified as a separate object, #5012.